front/components/actions/tables_query/TablesQueryActionDetails.tsx

```diff
@@ -92,78 +93,64 @@ function QueryThinking({ action }: { action: TablesQueryActionType }) {
   );
 }
 
-type TablesQueryActionWithResultsType = TablesQueryActionType & {
-  output: {
-    query_title?: string;
-    results: unknown[];
-  };
-};
-
-function hasQueryResults(
-  action: TablesQueryActionType
-): action is TablesQueryActionWithResultsType {
-  const { output } = action;
-
-  return (
-    output !== null &&
-    typeof output === "object" &&
-    "results" in output &&
-    Array.isArray(output.results)
-  );
-}
-
-function QueryTablesResults({ action }: { action: TablesQueryActionType }) {
+function QueryTablesResults({
+  action,
+  owner,
+}: {
+  action: TablesQueryActionType;
+  owner: WorkspaceType;
+}) {
   const sendNotification = useContext(SendNotificationsContext);
-
-  const handleDownload = useCallback(
-    (title: string) => {
-      if (!hasQueryResults(action)) {
-        return null;
-      }
-      if (
-        !action.output ||
-        !action.output.results ||
-        action.output.results.length === 0
-      ) {
-        return;
+  const { output } = action;
+  const title = getTablesQueryResultsFileTitle({ output });
+
+  const handleDownload = useCallback(() => {
+    if (action.resultsFileId) {
+      try {
+        const downloadUrl = `/api/w/${owner.sId}/files/${action.resultsFileId}?action=download`;
+        // Open the download URL in a new tab/window. Otherwise we get a CORS error due to the reirection
+        // to cloud storage.
+        window.open(downloadUrl, "_blank");
+      } catch (error) {
+        console.error("Download failed:", error);
+        sendNotification({
+          title: "Download Failed",
+          type: "error",
+          description: "An error occurred while opening the download link.",
+        });
       }
-
-      stringify(action.output.results, { header: true }, (err, csvOutput) => {
-        if (err) {
-          sendNotification({
-            title: "Error Downloading CSV",
-            type: "error",
-            description: `An error occurred while downloading the CSV: ${err}`,
-          });
-          return;
-        }
-
-        const blob = new Blob([csvOutput], { type: "text/csv" });
-        const url = URL.createObjectURL(blob);
-        const a = document.createElement("a");
-        a.href = url;
-        a.download = `${title}.csv`;
-        a.click();
+    } else {
+      sendNotification({
+        title: "No Results Available",
+        type: "error",
+        description: "There are no results available to download.",
       });
-    },
-    [action, sendNotification]
-  );
+    }
+  }, [action.resultsFileId, sendNotification, owner.sId]);
 
-  if (!hasQueryResults(action)) {
+  if (!action.resultsFileId) {
     return null;
   }
-  const { output } = action;
-  const title = output?.query_title ?? "query_results";
 
   return (
     <div>
-      <Collapsible defaultOpen={true}>
+      <span className="text-sm font-bold text-slate-900">Results</span>
+      <div onClick={() => handleDownload()} className="py-2">
```

Comment:
No anonymous function wrapping, otherwise you lose the point of `useCallback`.
```suggestion
      <div onClick={handleDownload} className="py-2">
```

Reply:
Indeed my bad 👍